fractal_tasks_core/dev/lib_descriptions.py

```diff
@@ -0,0 +1,179 @@
+    """
+    # Read docstring (via ast)
+    module_path = Path(fractal_tasks_core.__file__).parent / executable
+    module_name = module_path.with_suffix("").name
+    tree = ast.parse(module_path.read_text())
+    function = next(
+        f
+        for f in ast.walk(tree)
+        if (isinstance(f, ast.FunctionDef) and f.name == module_name)
+    )
+
+    docstring = ast.get_docstring(function)
+    # Parse docstring (via docstring_parser) and prepare output
+    parsed_docstring = docparse(docstring)
+    descriptions = {
+        param.arg_name: param.description.replace("\n", " ")
+        for param in parsed_docstring.params
+    }
+    return descriptions
+
+
+def _include_args_descriptions_in_schema(*, schema, descriptions):
+    """
+    Merge the descriptions obtained via `_get_args_descriptions` into an
+    existing JSON Schema for task arguments.
+    """
+    new_schema = schema.copy()
+    new_properties = schema["properties"].copy()
+    for key, value in schema["properties"].items():
+        if "description" in value:
+            raise ValueError("Property already has description")
+        else:
+            if key in descriptions:
+                value["description"] = descriptions[key]
+            else:
+                value["description"] = "Missing description"
+            new_properties[key] = value
+    new_schema["properties"] = new_properties
+    return new_schema
+
+
+INNER_PYDANTIC_MODELS = {
+    "OmeroChannel": "lib_channels.py",
+    "Window": "lib_channels.py",
+    "Channel": "lib_input_models.py",
+    "NapariWorkflowsInput": "lib_input_models.py",
+    "NapariWorkflowsOutput": "lib_input_models.py",
+}
+
+
+def _get_attributes_models_descriptions(
+    models: dict[str, str] = INNER_PYDANTIC_MODELS
+) -> dict[str, dict[str, str]]:
+    """
+    Extract attribut descriptions for Pydantic models
+    """
+    descriptions = {}
+
+    for model, module in models.items():
+        # get the class
+        module_path = Path(fractal_tasks_core.__file__).parent / module
+        tree = ast.parse(module_path.read_text())
+        try:
+            _class = next(
+                c
+                for c in ast.walk(tree)
+                if (isinstance(c, ast.ClassDef) and c.name == model)
+            )
+            descriptions[model] = {}
+        except StopIteration:
+            raise ValueError(f"Model {module_path}::{model} not found.")
+
+        # extract attribute docstrings
+        var_name: str = ""
+        for node in _class.body:
+            if isinstance(node, ast.AnnAssign):
+                descriptions[model][node.target.id] = "Missing description"
+                var_name = node.target.id
+            else:
+                if isinstance(node, ast.Expr) and var_name:
+                    descriptions[model][var_name] = node.value.s
+                    var_name = ""
+
+    return descriptions
+
+
+def _include_attributs_descriptions_in_schema(*, schema, descriptions):
```

Comment:
Rename to _insert_class_attrs_descriptions

Reply:
Transform into `_insert_class_attrs_descriptions(*, schema, class_name, descriptions: dict[str, str])`